splink/comparison_level.py

```diff
@@ -640,6 +637,8 @@ def as_dict(self):
 
         if self._is_null_level:
             output["is_null_level"] = True
+
```

Comment:
If we add "column_name" to the settings dict, we need to also update in the jsonschema file. @RobinL Actually, since this is maybe a more significant API addition

Reply:
Good spot.  The general idea is to keep this as minimal as possible, so if we can avoid adding the parent column name to the comparison level (and hence also adding to jsonschema), I think this would be desirable.

I haven't looked in detail so there may well be a compelling reason for doing this.  All I'm saying is we should only do this if there's a good reason!

Reply:
So, it looks to me as if we only really need to transform the null levels. If you think it better, we can go down the route you suggested above (feel free to make the change and PR).

Another alternative that I've added to this PR is to use sqlglot to parse the NULL sql condition and return a SQL string that can double up as a label name.

The function for this is [here](https://github.com/moj-analytical-services/splink/blob/comparison_level_composition/splink/sql_transform.py#L56).

You can see the tests and expected outputs from this functions [here](https://github.com/moj-analytical-services/splink/blob/comparison_level_composition/tests/test_sql_transform.py).